broken-site/broken-site-impl/src/main/java/com/duckduckgo/brokensite/impl/RealBrokenSitePrompt.kt

```diff
@@ -88,15 +81,33 @@ class RealBrokenSitePrompt @Inject constructor(
     }
 
     override suspend fun shouldShowBrokenSitePrompt(url: String): Boolean {
-        return isFeatureEnabled() &&
-            getUserRefreshesCount() >= REFRESH_COUNT_LIMIT &&
-            brokenSiteReportRepository.getNextShownDate()?.isBefore(currentTimeProvider.localDateTimeNow()) ?: true &&
-            !duckGoUrlDetector.isDuckDuckGoUrl(url)
+        if (!isFeatureEnabled() || getUserRefreshesCount() < REFRESH_COUNT_LIMIT || duckGoUrlDetector.isDuckDuckGoUrl(url)) {
+            return false
+        }
+
+        val currentTimestamp = currentTimeProvider.localDateTimeNow()
+
+        // Check if we're still in a cooldown period
+        brokenSiteReportRepository.getNextShownDate()?.let { nextDate ->
+            if (currentTimestamp.isBefore(nextDate)) {
+                return false
+            }
+        }
+
+        val dismissStreakResetDays = brokenSiteReportRepository.getDismissStreakResetDays().toLong()
+        val dismissalCount = brokenSiteReportRepository.getDismissalCountBetween(
+            currentTimestamp.minusDays(dismissStreakResetDays),
+            currentTimestamp,
+        )
+
+        return dismissalCount < brokenSiteReportRepository.getMaxDismissStreak()
     }
 
     override suspend fun ctaShown() {
+        val currentTimestamp = currentTimeProvider.localDateTimeNow()
         val nextShownDate = brokenSiteReportRepository.getNextShownDate()
-        val newNextShownDate = currentTimeProvider.localDateTimeNow().plusDays(brokenSiteReportRepository.getCoolDownDays())
+        val newNextShownDate = currentTimestamp.plusDays(brokenSiteReportRepository.getCoolDownDays())
+
         if (nextShownDate == null || newNextShownDate.isAfter(nextShownDate)) {
```

Comment:
nit: I think we can remove this check now and always store `newNextShownDate`, as with the current changes, `newNextShownDate` will always be after `nextShownDate`. Then, we can also remove `val nextShownDate = brokenSiteReportRepository.getNextShownDate()`